### PR TITLE
feat/provider: read .scwrc

### LINKED ISSUE #104: Feature Request: Read Organization ID and API Token from `~/.scwrc`
### Terraform Version
Terraform v0.11.11
+ provider.aws v1.51.0
+ provider.random v2.0.0
+ provider.scaleway v1.8.0

### Affected Resource(s)
N/A

### Terraform Configuration Files
N/A

### Debug Output
N/A

### Panic Output
N/A

### Expected Behaviour
`aws-cli` stores credentials in `~/.aws/credentials`. Amazon Web Services provider will try to read api key from it if no api key is provided in the configuration. Scaleway command line interface(`scw`) uses a similar file(`.scwrc`) to store credentials. Scaleway Provider should try to read credentials from that file as a fallback option.

### Actual Behaviour
Terraform will prompt me to input an organization id and an api key every time I run `terraform plan`.

### Steps to Reproduce
`terraform plan`

### Important Factoids
N/A

### References
- https://www.terraform.io/docs/providers/aws/index.html#shared-credentials-file

### PR DESCRIPTION
**Pull request by nicolai86**
once merged, the provider will fallback to scaleways ~/.scwrc to get
access credentials.

closes #104

### CODE REVIEW

File: scaleway/provider.go [hunk 1]
```diff
@@ -78,8 +99,20 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		}
 	}
 
+	organization := d.Get("organization").(string)
+
+	if apiKey == "" {
+		if path, err := homedir.Expand(d.Get(".scwrc").(string)); err == nil {
+			scwAPIKey, scwOrganization, err := readScalewayConfig(path)
+			if err == nil {
```
Comment (tombuildsstuff) [inline scaleway/provider.go]: should we invert this and raise this error?
```suggestion
			if err != nil {
			  return fmt.Errorf("Error loading credentials from SCW: %s", err)
			}
```

File: scaleway/provider.go [hunk 2]
```diff
@@ -78,8 +99,20 @@ func providerConfigure(d *schema.ResourceData) (interface{}, error) {
 		}
 	}
 
+	organization := d.Get("organization").(string)
+
+	if apiKey == "" {
```
Comment (tombuildsstuff) [inline scaleway/provider.go]: could we document this functionality [on the provider homepage](https://github.com/terraform-providers/terraform-provider-scaleway/blame/master/website/docs/index.html.markdown#L78)

File: scaleway/provider.go [hunk 3]
```diff
@@ -68,6 +71,24 @@ func Provider() terraform.ResourceProvider {
...
+
+func readScalewayConfig(path string) (string, string, error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return "", "", err
+	}
+	defer f.Close()
+
+	var data scalewayConfig
+	err = json.NewDecoder(f).Decode(&data)
```
[the remark this answers is not in the document]
Comment (nicolai86) [reply scaleway/provider.go]: good point, I'll update this

File: scaleway/provider.go (hunk 1 above)
Comment (nicolai86) [reply scaleway/provider.go]: that's a miss, yes it should be inverted

File: scaleway/provider.go (hunk 2 above)
Comment (nicolai86) [reply scaleway/provider.go]: done!

**Comment by nicolai86** [comment]
Ran all the tests over night, without setting environment variables - they all passed. Will pull this in :)

**Pull request merged**